src/sdk/services/blockchain/contracts/VestingContract.ts

```diff
@@ -231,4 +231,58 @@ export class VestingContract {
 
         return totalBalance;
     }
+
+    getVestingCategory(categoryId: number): {
+        startDelay: number;
+        cliffPeriod: number;
+        vestingPeriod: number;
+        tgeUnlock: number;
+    } {
+        const vestingCategory = vestingCategories.get(categoryId);
+
+        if (!vestingCategory) {
+            throw new Error(`Vesting category ${categoryId} not found`);
+        }
+
+        return vestingCategory;
+    }
+
+    async getUnlockableBalance(account: NameType): Promise<number> {
+        const allocations = await this.getAllocations(account); // Fetch all allocations for the account
+        let totalUnlockable = 0;
+
+        const currentTime = Math.floor(Date.now() / 1000); // Current time in seconds
+
+        for (const allocation of allocations) {
+            const tokensAllocated = parseFloat(allocation.tokens_allocated.split(' ')[0]);
+            const tokensClaimed = parseFloat(allocation.tokens_claimed.split(' ')[0]);
+
+            // Extract vesting details
+            const categoryId = allocation.vesting_category_type;
+            const vestingCategory = this.getVestingCategory(categoryId);
```

Comment:
need to add the other vesting categories to `vestingCategories` so it can fetch all of them